napari/components/viewer_model.py

```diff
@@ -39,47 +44,32 @@ class ViewerModel(AddLayersMixin, KeymapHandler, KeymapProvider):
         Preset color palettes.
     """
 
-    themes = palettes
-
-    def __init__(
-        self, title='napari', ndisplay=2, order=None, axis_labels=None
-    ):
-        super().__init__()
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            status=Event,
-            help=Event,
-            title=Event,
-            interactive=Event,
-            cursor=Event,
-            reset_view=Event,
-            active_layer=Event,
-            palette=Event,
-            grid=Event,
-            layers_change=Event,
-        )
-
+    title: str = 'napari'
+    status: str = 'Ready'
+    help: str = ''
+    cursor: str = 'standard'
+    cursor_size: Optional[int] = None
+    interactive: bool = True
+    active_layer: Optional[int] = None
+    grid_size: Tuple[int, int] = (1, 1)
+    theme: str = DEFAULT_THEME
+
+    grid_stride: ClassVar[int] = 1
+    palette: ClassVar[Dict[str, str]] = palettes[DEFAULT_THEME]
+    layers: ClassVar[LayerList] = []
+
+    ndisplay: InitVar[int] = 2  # type: ignore
+    order: InitVar[Optional[Tuple[int, ...]]] = None  # type: ignore
+    axis_labels: InitVar[Optional[Sequence[str]]] = None  # type: ignore
+
+    def __post_init__(self, ndisplay, order, axis_labels):
+        super().__init__()  # will this work?
```

Comment:
I guess we needed the `super().__init__()` call because of our inheritance pattern above!! Pretty crazy if all that inheritance worked just fine with the dataclass approach and didn't need to be changed.

We're we adding attributes in those base classes? I could see how that would cause problems with the dataclass syntax, but I guess if we were just adding methods it might be ok

Reply:
yeah, it's working!  I _think_ we're just adding methods... but I need to check.  If we were also adding attributes, it can sometimes still be ok, but it's a little more tricky.